src/core/lib/security/credentials/tls/grpc_tls_credentials_options.h

```diff
@@ -126,6 +126,25 @@ struct grpc_tls_credentials_options
   // not enable CRL checking. Only supported for OpenSSL version > 1.1.
   void set_crl_directory(std::string path) { crl_directory_ = std::move(path); }
 
+  bool operator==(const grpc_tls_credentials_options& other) const {
+    return cert_request_type_ == other.cert_request_type_ &&
+           verify_server_cert_ == other.verify_server_cert_ &&
+           min_tls_version_ == other.min_tls_version_ &&
+           max_tls_version_ == other.max_tls_version_ &&
+           (verifier_ == other.verifier_ ||
+            (verifier_ != nullptr && other.verifier_ != nullptr &&
+             verifier_->cmp(other.verifier_.get()) == 0)) &&
+           check_call_host_ == other.check_call_host_ &&
+           provider_ == other.provider_ &&
```

Comment:
Why are we plumbing the `cmp()` method into the verifier but not into the provider?  Seems like we should be consistent here: either we always require both to be the same instance, or we require both to compare the same.

I'd be fine with either approach.  Requiring them to be the same instance is simpler, because it eliminates the need to plumb the `cmp()` method into the verifier API.  But I'd also be fine with plumbing the `cmp()` method into the provider as well.

Reply:
We need it in verifier because of an explicit need in XdsCertificateVerifier https://github.com/grpc/grpc/blob/87068e02c3c4d34f51019b3ccdceea1230c92fea/src/core/lib/security/credentials/xds/xds_credentials.cc#L163.. We don't have such a need for providers yet.

Reply:
Okay, then let's do the same thing for the provider.  I'd prefer to be consistent here.  Just because we don't need it yet doesn't mean the infrastructure should not be in place for it.

Reply:
Added. The implementation of those comparators might have consequences that I don't want to deal with here, so leaving those as TODOs.